packages/oas/test/operation/index.test.ts

```diff
@@ -1051,127 +1052,184 @@ describe('#getOperationId()', () => {
-      ],
-      [
-        'should clean up an operationId that starts with a number',
-        {
-          operationId: '400oD_Browse_by_Date_Feed',
-          expected: '_400oD_Browse_by_Date_Feed',
-        },
-      ],
-    ])('%s', (_, { operationId, expected }) => {
-      const spec = Oas.init({
-        openapi: '3.1.0',
-        info: {
-          title: 'testing',
-          version: '1.0.0',
-        },
-        paths: {
+    describe('given an operationId that contains non-alphanumeric characters', () => {
+      it.each([
+        ['camelCase', 'findPetsByStatus'],
+        ['friendlyCase', 'findPetsByStatus'],
+      ])('%s', (option, expected) => {
+        const spec = createOasForPaths({
           '/anything': {
             get: {
-              operationId,
+              // This mess of a string is intentionally nasty so we can be sure that we're not
+              // including anything that wouldn't look right as an operationID for a potential method
+              // accessor in `api`.
+              operationId: 'find/?*!@#$%^&*()-=.,<>+[]{}\\|pets-by-status',
+              responses: {},
             },
           },
-        },
+        });
+
+        const operation = spec.operation('/anything', 'get');
+        expect(operation.getOperationId({ [option]: true })).toBe(expected);
+      });
+    });
+
+    describe('given an operationId that ends in non-alphanumeric characters', () => {
+      it.each([
+        ['camelCase', 'findPetsByStatusDeprecated'],
+        ['friendlyCase', 'findPetsByStatusDeprecated'],
+      ])('%s', (option, expected) => {
+        const spec = createOasForPaths({
+          '/anything': {
+            get: {
+              operationId: 'find pets by status (deprecated?*!@#$%^&*()-=.,<>+[]{})',
+              responses: {},
+            },
+          },
+        });
+
+        const operation = spec.operation('/anything', 'get');
+        expect(operation.getOperationId({ [option]: true })).toBe(expected);
+      });
+    });
+
+    describe('given an operationId that begins with non-alphanumeric characters', () => {
+      it.each([
+        ['camelCase', 'findPetsByStatus'],
+        ['friendlyCase', 'findPetsByStatus'],
+      ])('%s', (option, expected) => {
+        const spec = createOasForPaths({
+          '/anything': {
+            get: {
+              operationId: '(?*!@#$%^&*()-=.,<>+[]{}) find pets by status',
+              responses: {},
+            },
+          },
+        });
+
+        const operation = spec.operation('/anything', 'get');
+        expect(operation.getOperationId({ [option]: true })).toBe(expected);
       });
+    });
 
-      const operation = spec.operation('/anything', 'get');
-      expect(operation.getOperationId({ camelCase: true })).toBe(expected);
+    describe('given an operationId that begins with a number', () => {
+      it.each([
+        ['camelCase', '_400oD_Browse_by_Date_Feed'],
+        ['friendlyCase', '_400oDBrowseByDateFeed'],
```

Comment:
just curious, is there a reason we're preceding these operationId's with an underscore?

Reply:
These are eventually used as method names in `api` and JS methods can't begin with a number.

Reply:
ah, got it!